### DISCUSSION TITLE
addition of polychromatic field at far field

### DISCUSSION THREAD

**Discussion by XLIU-1993** [rafael-fuente/Diffraction-Simulations--Angular-Spectrum-Method#27] [Q&A]
Dear Rafeal,

Thank you for this fantastic and very intuitive work! I read through some of your codes, but I still didn't figure out how are the polychromatic fields added at the far field. I mean for different wavelengths, the propagated wavefield is zoomed in with a rate of lambda1/lambda2, could you please give some details?

Best,
Xu

**Comment by rafael-fuente** [comment]
Hello,

Thank you! 
Polychromatic light is added, simulating the field propagation for different wavelengths; there isn't implemented 'artificial zoom'; everything comes naturally from the optical wave equations, and the relation zoom = cte = lambda1/lambda2 is not generally valid (only for the paraxial far field). 

The total field is the coherent sum of all single wavelength fields, but for computing the colors, the spectral radiance is integrated to get the chromatic CIE tristimulus values.

**Reply by XLIU-1993** [reply]
Hello Rafael,

I didn't get it. Maybe I can describe the farfield diffraction process more clearly in this way. In this situation, I will only consider the incoming wave field (E), the object (O), and the camera plane (Cam), the distance between Cam and O is (Z).

Normally, I should predefine the pixel size (dx) and the pixel number (N) of the object (O), and the electric field(E), then Psi is the exit wave field at the backside of the object Psi=O*E,  before doing FFT, I need to calculate the pixel size(dwx) at the camera plan which at far field should be like: dwx = wavelength*Z/(N*dx), from here, I see that the different wavelength results in different pixel size at the camera plan, how could I sum the diffraction patterns without 'artificial zoom'?

There is another thing that confuses me. If I considered this process inversely, the predefined pixel size of the object is not realistic, because different wavelength 'sees' different pixel size at the object plane.

Do you have some insights?
Have a  good day!
Xu
